### ISSUE TITLE
New model deepgrow_v2: with multi-head inference 

### ISSUE THREAD

**Issue by AHarouni** [Project-MONAI/MONAILabel#1328]
**Is your feature request related to a problem? Please describe.**

Need to be able to train a unet like model (segresnet,etc) as it exist today Plus in addition add a second head to produce 2 outputs. forground and background which would work as deepgrow mode

This network would be trained in 2 modes:
- Main network mode. train segresnet as is with normal 10 labels head
- Deep grow mode. we need to freeze all or segresnet layer. only allow to train the new second head taking from the encoding / decoding and have a new FC layer with 2 outputs 

This feature request is related to the following bugs:
- [Finetune new model by freezing all layers except FC](https://github.com/Project-MONAI/MONAILabel/issues/1298)
- [Deepedit has low resolution and non responsive to new clicks](https://github.com/Project-MONAI/MONAILabel/issues/1299)

**Describe the solution you'd like**
Provide new network architecture with 2 heads. This would allow us to train the same core encoding and decoding layer or the network

Allow for 2 modes of training:
- train all label
- train deepgrow
Allow 2 modes to run infer:
- Run infer as normal segmentation
- Run deep grow 

**Describe alternatives you've considered**

```
checkptPath = "/rootpath/train_01/train_model.pt"
checkpoint = torch.load(checkptPath)
model_state_dict = checkpoint.get(self.model_state_dict, checkpoint)
self.network.load_state_dict(model_state_dict, strict=False)
```

**Comment by SachidanandAlle** [comment]
DeepEdit or Deepgrow?

**Comment by AHarouni** [comment]
This ask is to create a new network architecture that is similar to deepgrow but can also run segmentation.
